### PR TITLE
feat: Add a bind_cached_fk to the BaseModel

### PR DESCRIPTION
**Pull request by mattrobenolt**
Model.bind_cached_fk provides a way that we can freely assure that
traversing a foreign key relationship is done in the most efficient way
possible.

* If the relationship has already been bound, it does nothing.
* If not bound, it checks if the model can be fetched using
get_from_cache.
* Falls back to a normal get and query from the database.

This pattern allows things such as:

group.bind_cached_fk('project').project

And make sure this is done the best we can.

### CODE REVIEW

File: src/sentry/db/models/base.py [hunk 1]
```diff
@@ -117,6 +117,37 @@ def old_value(self, field_name):
             return None
         return self.__data.get(field_name)
 
+    def bind_cached_fk(self, *rels):
```
Comment (tkaemming) [inline src/sentry/db/models/base.py]: Why isn't `rels` just a list? Passing it as a variadic makes easier to accidentally turn it into a noop by not passing any arguments, and limits the potential for adding more arguments in the future that don't have to be plucked out of `**kwargs`.

**Review by tkaemming — CHANGES_REQUESTED** [review]
Seems like it shouldn't be hard to wire up at least a trivial test to this to ensure there's _some_ coverage.

**Comment by BYK** [comment]
Closing due to staleness.

**Comment by mattrobenolt** [comment]
RIP, this was a nice idea.

**Pull request closed without merging**